test/test_helper.hpp

```diff
@@ -2,264 +2,28 @@
-
-        static void reset() { counter() = 0; }
-
-        static std::atomic_int& counter() {
-            static std::atomic_int counter;
-            return counter;
-        }
-    private:
-        const size_t _id = no; // only used for debugging purpose
-    };
-
-
-    class test_exception : public std::exception {
-        const std::string _error;
-
-    public:
-        test_exception() {}
-
-        explicit test_exception(const std::string& error);
-
-        explicit test_exception(const char* error);
-
-        test_exception& operator=(const test_exception&) = default;
-        test_exception(const test_exception&) = default;
-        test_exception& operator=(test_exception&&) = default;
-        test_exception(test_exception&&) = default;
-
-        virtual ~test_exception() {}
-
-        const char* what() const noexcept override;
-    };
-
-
-    struct test_setup
-    {
-        test_setup() {
-            custom_scheduler<0>::reset();
-            custom_scheduler<1>::reset();
-        }
-    };
-
-    template <typename T>
-    struct test_fixture
-    {
-        test_fixture()
-            : _task_counter{ 0 } {
-            custom_scheduler<0>::reset();
-            custom_scheduler<1>::reset();
-        }
-
-        ~test_fixture() {}
-
-        stlab::future<T> sut;
-
-        template <typename... F>
-        void wait_until_future_completed(F&... f) {
-            (void)std::initializer_list<int>{ (wait_until_future_is_ready(f), 0)... };
-        }
-
-        template <typename F>
-        auto wait_until_future_r_completed(F& f) {
-            auto result = f.get_try();
-            while (!result.is_initialized()) {
-                result = f.get_try();
-            }
-            return std::move(result);
-        }
-
-        void check_valid_future() {}
-
-        void check_valid_future(const stlab::future<T>& f) {
-            BOOST_REQUIRE(f.valid() == true);
-            BOOST_REQUIRE(f.error().is_initialized() == false);
-        }
-
-        template <typename F, typename... FS>
-        void check_valid_future(const F& f, const FS&... fs) {
-            BOOST_REQUIRE(f.valid() == true);
-            BOOST_REQUIRE(f.error().is_initialized() == false);
-            check_valid_future(fs...);
-        }
-
-        template <typename E, typename F>
-        static void check_failure(F& f, const char* message) {
-            BOOST_REQUIRE_EXCEPTION(f.get_try(), E, ([_m = message](const auto& e) { return std::string(_m) == std::string(e.what()); }));
-        }
-
-        template <typename E, typename... F>
-        void wait_until_future_fails(F&... f) {
-            (void)std::initializer_list<int>{ (wait_until_this_future_fails<E>(f), 0)... };
-        }
-
-        void wait_until_all_tasks_completed() {
-            while (_task_counter.load() != 0) { std::this_thread::sleep_for(std::chrono::milliseconds(1)); }
-        }
-
-
-        std::atomic_int _task_counter;
-
+    class move_only_t {
```

Comment:
It would be great if this type could be moved to `stlab/test/model.hpp`. It would be useful for testing e.g., `task`.